warden/x/warden/keeper/genesis.go

```diff
@@ -22,6 +22,13 @@ func (k *Keeper) ImportState(ctx sdk.Context, genState types.GenesisState) error
 		return fmt.Errorf("failed to import spaces: %w", err)
 	}
 
+	for _, key := range genState.Spaces {
+		err := k.SpacesKeeper.updateSpaceOwners(ctx, key, nil)
+		if err != nil {
+			return fmt.Errorf("failed to import space owners: %w", err)
+		}
+	}
```

Comment:
maybe one thing we could do is to put this inside a new method `k.SpacesKeeper.Import(...)` that inside calls keeps the index updates.

so we don't leak the spaceOwners index here

Reply:
Thank you for your comments, changed as requested